clients/drcachesim/tests/prefetch-aarch64-basic-counts.templatex

```diff
@@ -0,0 +1,32 @@
+Hello, world!
+---- <application exited with code 0> ----
+Basic counts tool results:
+Total counts:
+     .* total \(fetched\) instructions
+     .* total unique \(fetched\) instructions
+     .* total non-fetched instructions
+          62 total prefetches
+     .* total data loads
+     .* total data stores
+           1 total threads
+     .* total scheduling markers
+     .* total transfer markers
+     .* total function id markers
+     .* total function return address markers
+     .* total function argument markers
+     .* total function return value markers
+     .* total other markers
+Thread .* counts:
+     .* \(fetched\) instructions
+     .* unique \(fetched\) instructions
+     .* non-fetched instructions
+          62 prefetches
```

Comment:
Some 26 prefetches occur independently of the prefetches issued explicitly in the test. 

This makes the total expected prefetches in prefetch-aarch64-basic-counts = 36+26 = 62.

Also, in the other test (prefetch-basic-counts), this makes the total expected prefetches = 8+26 = 34

Reply:
> Some 26 prefetches occur independently of the prefetches issued explicitly in the test.

That is going to make this a flaky test as it is likely quite dependent on the particular compiler and/or system libraries.

There are 3 potential solutions:
1) Make the test app pure asm so there is no compiler-added or system library code.  This means it won't run on Windows -- but we can live with that.
2) Use a checked-in offline trace (with no online test).
3) Make the regex handle variations.

Reply:
I used a combination of (2) and (3). I modified the existing regex to verify only that prefetch instruction count is non-zero. And I also checked-in an offline trace collected on AArch64 that verifies the exact values.

About (1), I tried using a very simple asm code, but even that contains some 14 prefetches. Were you suggesting a different way?
```
abhinavas@tx1:~/dr/build/i4328-1$ cat a.s
.global main
main:
abhinavas@tx1:~/dr/build/i4328-1$ gcc -O0 a.s
abhinavas@tx1:~/dr/build/i4328-1$ ./a.out
abhinavas@tx1:~/dr/build/i4328-1$ rm -r drmemtrace.*;  ./bin64/drrun -t drcachesim  -offline ./a.out; ./bin64/drrun -t drcachesim -indir drmemtrace* -simulator_type basic_counts
Basic counts tool results:
Total counts:
       69762 total (fetched) instructions
        6531 total unique (fetched) instructions
           0 total non-fetched instructions
          14 total prefetches
       17886 total data loads
        7626 total data stores
           1 total threads
          56 total scheduling markers
           0 total transfer markers
           0 total function id markers
           0 total function return address markers
           0 total function argument markers
           0 total function return value markers
           1 total other markers
Thread 11801 counts:
       69762 (fetched) instructions
        6531 unique (fetched) instructions
           0 non-fetched instructions
          14 prefetches
       17886 data loads
        7626 data stores
          56 scheduling markers
           0 transfer markers
           0 function id markers
           0 function return address markers
           0 function argument markers
           0 function return value markers
           1 other markers
abhinavas@tx1:~/dr/build/i4328-1$ 
```

Reply:
> About (1), I tried using a very simple asm code, but even that contains some 14 prefetches. Were you suggesting a different way?

Build with absolutely no startup code so the *only* code is the asm code and there will be nothing else but the precise instructions you place there.  See e.g. suite/tests/common/allasm_aarch64_cache.asm and related files which are built `-nostartfiles -nodefaultlibs -static`.

Reply:
Thanks. I was able to build some asm code without any of those 'default' prefetches.

In this PR, do you suggest we switch to a pure asm test app suggested in (1), instead of the tests added currently?

Reply:
I think a pure asm test is superior to a regex checking for non-zero.  You could also have a custom-analysis test that checks for the different types of prefetches.  Another method of controlling the precise count is to use attach+detach around the traced code.

Reply:
attach+detach seems interesting, but based on the latest update on #1551, I think start/stop support is not implemented yet for ARM. I'll add a pure asm test instead.

About the custom analysis test, this will be only for offline trace right? To use it for online, we'll need to add it as another simulator_type in drcachesim. I'm assuming we don't want to do that only for this test.

Reply:
Added a custom analyzer test for a pure asm app that checks each of the new prefetch trace types.